lib/output.js

```diff
@@ -784,7 +784,9 @@ function OutputStream(options) {
         output.with_parens(function(){
             self.condition.print(output);
         });
-        output.semicolon();
+        if (output.option("beautify") && output.option("screw_ie8")) {
+            output.semicolon();
```

Comment:
Indeed - before the if-do-while-else case would always work on IE8 regardless of `--support-ie8`

I guess we'll find out how big a surprise this would be after the next release.